Ray.jl/src/runtime.jl

```diff
@@ -181,16 +175,21 @@ function task_executor(ray_function, returns_ptr, task_args_ptr, task_name,
                                 rayjll.unwrap_function_descriptor(fd),
                                 get_current_job_id())
         # TODO: write generic Ray.put and Ray.get functions and abstract over this buffer stuff
-        args = map(task_args) do arg
+        flattened = map(task_args) do arg
             v = take!(rayjll.GetData(arg[]))
             io = IOBuffer(v)
             return deserialize(io)
         end
+        args, kwargs = recover_args(flattened)
 
-        arg_string = join(string.("::", typeof.(args)), ", ")
-        @info "Calling $func($arg_string)"
+        @info begin
+            args_str = join([string("::", typeof(arg)) for arg in args], ", ")
+            kwargs_str = join([string(k, "::", typeof(v)) for (k, v) in kwargs], ", ")
```

Comment:
I liked the broadcasting but maybe it's too clever?  or doesn't work with the kv pairs I guess...  if you're gonna use generators might as well make them fully lazy to not allocate teh intermediate arrays

```suggestion
            args_str = join((string("::", typeof(arg)) for arg in args), ", ")
            kwargs_str = join((string(k, "::", typeof(v)) for (k, v) in kwargs), ", ")
```